### PR TITLE
feat(table): Add component tokens

### PR DESCRIPTION
**Pull request by macandcheese**
**Related Issue:** #7180 

## Summary
Screenshot 2024-03-14 at 3 51 21 PM

### CODE REVIEW

File: packages/calcite-components/src/components/table-row/table-row.scss [hunk 1]
```diff
@@ -21,9 +24,23 @@
   }
 }
 
+:host calcite-table-cell {
```
Comment (alisonailea) [inline packages/calcite-components/src/components/table-row/table-row.scss:27]: the :host here isn't adding anything and can be removed.

File: packages/calcite-components/src/components/table-row/table-row.scss [hunk 2]
```diff
@@ -21,9 +24,23 @@
...
+  --calcite-internal-table-cell-background-color: var(
+    --calcite-table-row-background-color,
+    var(--calcite-color-foreground-1)
+  );
+}
+
+:host([selected]) calcite-table-cell {
+  --calcite-table-cell-background-color: var(
```
Comment (alisonailea) [inline packages/calcite-components/src/components/table-row/table-row.scss:35]: I think this should be `--calcite-internal-table-cell-background-color` ?

**Review by alisonailea — CHANGES_REQUESTED** [review]
Looks good! Just a few little things.

File: packages/calcite-components/src/components/table-cell/table-cell.scss [hunk 3]
```diff
@@ -42,44 +54,67 @@ td.last-cell {
...
 }
 
-.number-cell {
-  @apply bg-foreground-2;
-}
-
 .footer-cell {
-  @apply bg-background font-medium text-color-1;
-  border-block-start: 1px solid var(--calcite-color-border-3);
+  @apply font-medium text-color-1;
```
Comment (alisonailea) [inline packages/calcite-components/src/components/table-cell/table-cell.scss:58]: should this "text-color-1" have a CSS var/token?

**Review by alisonailea — APPROVED** [review]
Looks great! One question but approved 🎉

File: packages/calcite-components/src/components/table-cell/table-cell.scss (hunk 3 above)
Comment (macandcheese) [reply packages/calcite-components/src/components/table-cell/table-cell.scss:58]: Yep, will update

**Pull request merged**